### DISCUSSION TITLE
Using Viewflow with existing Django Functional Based Views

### DISCUSSION THREAD

**Discussion by ranganath294** [viewflow/viewflow#451] [General]
I have a Django backend ready and i want to integrate viewflow to it. I have used only functional views in my application. Also all of them are apis. I have my own frontend and i dont want to use viewflow's ui. Is it possible to integrate viewflow with these conditions?
My application contains multiple user roles, various set of permissions for different user groups and also object level permissions with Django Guardain. It has a parallel workflow. Also i want to use workflow explicitly (a seperate model to maintain the workflow of my application if i am not wrong). And it has a bit dynamic nature......Is it possible to integrate viewflow to my application considering i am using functional based views (apis) ???

**Comment by kmmbvnr** [comment]
Sure, you can integrate Viewflow with your Django backend using functional views and APIs. You don't need to use Viewflow's UI.

For integrating with functional views, check this guide: [Viewflow Custom Views](https://docs.viewflow.io/workflow/custom_views.html#function-based-views).

To start without Viewflow's attached views, simply change your import from viewflow.workflow import flow to from viewflow.workflow import nodes as flow and write your flow as usual.

**Reply by ranganath294** [reply]
Is there any example demo project for this type of use case ?? It would be helpful. In the pause branch of cookbook, there is a NoFrontend but it is not available in the repository.

Also, if there is any other project to understand viewflow, please tell where to find it ?

**Reply by kmmbvnr** [reply]
All examples related to the open-sourced functionality of Viewflow can be found in the cookbook. The best way to learn Viewflow is by working through the documentation's quick start tutorials, focusing on the sections most relevant to your needs.

For a no-frontend setup, start with `from viewflow.workflow import nodes as flow` and then proceed as usual. However, you will need to create your own views for everything.

**Comment by ranganath294** [comment]
Do the free version features remain free or is there a possibility of them moved to paid version because once after using these features, if they will be removed, then it will be a tedious task to make the app work

**Reply by kmmbvnr** [reply]
You raised an important question. I realized that the subprocess nodes were accidentally included in the open-source version and will need to be removed, along with some advanced CRUD features. The best way to ensure sustainable project development is by purchasing software updates. However, the version you currently have will remain unchanged.
